apps/directory/src/functions/bookmarkMapper.js

```diff
@@ -0,0 +1,133 @@
+import router from "../router";
+import { useFiltersStore } from '../stores/filtersStore'
+import { useCollectionStore } from '../stores/collectionStore';
+import { useCheckoutStore } from '../stores/checkoutStore';
+
+function setBookmark (bookmark) {
+    router.push(
+        {
+            name: router.currentRoute.name,
+            query: bookmark
+        },
+    )
+}
+
+export async function applyBookmark (watchedQuery) {
+    const checkoutStore = useCheckoutStore();
+    const collectionStore = useCollectionStore();
+    const filtersStore = useFiltersStore();
+
+    let query = watchedQuery;
+
+    if (!query) {
+        const route = router.currentRoute.value
+
+        if (!route.query) return
+
+        query = route.query
+    }
+
+    if (!query || !Object.keys(query).length > 0) return
+
+    /**  negotiator token */
+    // if (query.nToken) {
+    //   state.nToken = query.nToken
+    // }
```

Comment:
remove

Reply:
no, it is for the next step :) keeps me from going back and forth between two repos